src/thx/promise/Promise.hx

```diff
@@ -233,7 +236,7 @@ abstract Promise<T>(Future<Result<T, Error>>) to Future<Result<T, Error>> {
 }
 
 class Promises {
-  public static function join<T1,T2>(p1 : Promise<T1>, p2 : Promise<T2>) : Promise<Tuple2<T1,T2>> {
```

Comment:
It obviates the need for the joins, which are now just a special case, but there's no real need to deprecate.